### DISCUSSION TITLE
Exporting random segments of video?

### DISCUSSION THREAD

**Discussion by masonevans** [blakeblackshear/frigate#2226] [Ask A Question] [answered]
I use my cameras for a combination of security & wildlife videos.  There are lots of times where Frigate doesn't detect an object or create an event but I'd still like to save a random segment of video (for example, last night there was a bobcat walking around that wasn't detected as an object).

Is there an easy way to do this that anyone has found?

**Comment by blakeblackshear** [comment, marked as answer]
There is an endpoint for this, but it's not in the docs. https://github.com/blakeblackshear/frigate/blob/master/frigate/http.py#L559

**Reply by blakeblackshear** [reply]
Note that if the file is too large, it may fill your cache and crash frigate.

**Reply by masonevans** [reply]
Thanks, that worked!

For anyone reading this who doesn't know Python, you can hit the following URL to get the video

`http://<host>/api/<camera name>/start/<start ts>/end/<end ts>/clip.mp4`

Where `ts` is a Unix epoch, can be converted here: https://www.epochconverter.com

If you want to download the video instead of viewing it in the browser you can append `?download=true`

**Reply by vakilando** [reply]
> Better yet, if we could have a timeline

Yes a timeline woud be perfect!
I'm just "migrating" from xeoma to frigate and I am missing the timeline xeoma provides for the archive (records). In this timeline you see the events and can choose them in a calendar like dialogue (see the screenshot).
![xeoma-timeline](https://user-images.githubusercontent.com/36138919/149833908-774cb740-5e61-4818-af37-7879b1c9d1c2.png)
